app/models/Visitor.py

```diff
@@ -36,9 +40,15 @@ def add(cls):
         ip_addr = get_ip()
         visited_time = utcnow()
         visitor = Visitor(url=url, ip_addr=ip_addr, visited_time=visited_time)
-        db.session.add(visitor)
-        db.session.commit()
-        current_app.logger.info(url + ' is recorded')
+        
+        if len(cls.queue) < cls.capacity: 
```

Comment:
what is cls? should cls.queue be global?

Reply:
cls is Visitor. This is actually a static method.